DogScepterLib/Core/Models/GMObject.cs

```diff
@@ -1,196 +1,362 @@
-            writer.Write(MaskSpriteID);
-            writer.WriteWideBoolean(Physics);
-            writer.WriteWideBoolean(PhysicsSensor);
-            writer.Write((int)PhysicsShape);
-            writer.Write(PhysicsDensity);
-            writer.Write(PhysicsRestitution);
-            writer.Write(PhysicsGroup);
-            writer.Write(PhysicsLinearDamping);
-            writer.Write(PhysicsAngularDamping);
-            writer.Write(PhysicsVertices.Count);
-            writer.Write(PhysicsFriction);
-            writer.WriteWideBoolean(PhysicsAwake);
-            writer.WriteWideBoolean(PhysicsKinematic);
-            foreach (PhysicsVertex v in PhysicsVertices)
-                v.Serialize(writer);
-            Events.Serialize(writer);
+            writer.Write(X);
+            writer.Write(Y);
         }
 
         public void Deserialize(GMDataReader reader)
         {
-            Name = reader.ReadStringPointerObject();
-            SpriteID = reader.ReadInt32();
-            Visible = reader.ReadWideBoolean();
-            Solid = reader.ReadWideBoolean();
-            Depth = reader.ReadInt32();
-            Persistent = reader.ReadWideBoolean();
-            ParentObjectID = reader.ReadInt32();
-            MaskSpriteID = reader.ReadInt32();
-            Physics = reader.ReadWideBoolean();
-            PhysicsSensor = reader.ReadWideBoolean();
-            PhysicsShape = (CollisionShape)reader.ReadInt32();
-            PhysicsDensity = reader.ReadSingle();
-            PhysicsRestitution = reader.ReadSingle();
-            PhysicsGroup = reader.ReadInt32();
-            PhysicsLinearDamping = reader.ReadSingle();
-            PhysicsAngularDamping = reader.ReadSingle();
-            int vertexCount = reader.ReadInt32();
-            PhysicsFriction = reader.ReadSingle();
-            PhysicsAwake = reader.ReadWideBoolean();
-            PhysicsKinematic = reader.ReadWideBoolean();
-            PhysicsVertices = new List<PhysicsVertex>(vertexCount);
-            for (int i = vertexCount; i > 0; i--)
-            {
-                PhysicsVertex v = new PhysicsVertex();
-                v.Deserialize(reader);
-                PhysicsVertices.Add(v);
-            }
-            Events = new GMUniquePointerList<GMUniquePointerList<Event>>();
-            Events.Deserialize(reader);
+            X = reader.ReadSingle();
+            Y = reader.ReadSingle();
         }
+    }
 
-        public override string ToString()
+    /// <summary>
+    /// An event that a <see cref="GMObject"/> uses.
+    /// </summary>
+    public class Event : IGMSerializable
+    {
+        /// <summary>
+        /// The subtype of this event.
+        /// </summary>
+        /// <remarks>GameMaker suffixes <see cref="GMCode"/>.<see cref="GMCode.Name"/>, which
+        /// <see cref="Action"/>.<see cref="Action.CodeID"/> references, with this ID.</remarks>
+        //TODO: UMT has this as an UINT, while this here is an INT. Who's correct?
```

Comment:
I don't think the number gets high enough to matter, but it's most likely _technically_ a normal int.

Reply:
I don't think so, but I'm pretty sure the type is still just `int`.